File: reto1.py
def anagrama(a,b):
    if len(a) == len(b) and a != b:
        cont = 0
        lstA = sorted((list(a)))
        lstB = sorted((list(b)))
        for i in lstA:
            if i == lstB[lstA.index(i)]:
                cont +=1
        return cont == len(a) and cont == len(b)     
    else:
        return False

print(anagrama("alan", "lana"))

File: reto1.php
<?php
function anagrama($a,$b){
    $cont = 0;
    $igual = 0;
    if(strlen($a)==strlen($b) and $a != $b){
        $lstA = sort(str_split($a));
        $lstB = sort(str_split($b));
        foreach($lstA as $i){
            if($i == $lstB[$cont]){
                $igual +=1;
            }
            $cont +=1;
        }
    }
    return (strlen($a) == $igual and strlen($b)== $igual);
};
echo (anagrama("alan","lana"));
?>